src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1310,6 +1410,64 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  m_lock.Lock();
+  if (m_state == STATE_STOPPED) {
+
+    dout(20) << "replayer is already stopped, skip to delete image step of "
```

Comment:
Can we combine the logic here with the logic used in ```handle_stop``` so that we only have one helper method that unregisters the client, schedules the delete, etc?